pkg/recovery/etcd_template.go

```diff
@@ -0,0 +1,135 @@
+package recovery
+
+var RecoveryEtcdTemplate = []byte(`apiVersion: v1
+kind: Pod
+metadata:
+  name: recovery-etcd
+  namespace: kube-system
+  labels:
+    k8s-app: recovery-etcd
+spec:
+  initContainers:
+  - name: recovery
+    image: {{ .Image }}
+    command: 
+    - /usr/local/bin/etcdctl
+    - snapshot
+    - restore
+    - --data-dir=/var/etcd/recovery
+    - --name=recovery-etcd
+    - --initial-cluster=recovery-etcd=http://localhost:32380
+    - --initial-cluster-token=bootkube-recovery
+    - --initial-advertise-peer-urls=http://localhost:32380
+    - --skip-hash-check=true
+    - /var/etcd-backupdir/{{ .BackupFile }}
+    env:
+    - name: ETCDCTL_API
+      value: "3"
+    volumeMounts:
+      - mountPath: /var/etcd
+        name: etcd
+        readOnly: false
+      - mountPath: /var/etcd-backupdir
+        name: etcdbackup
+        readOnly: false
+  containers:
+  - name: etcd
+    image: {{ .Image }}
+    command:
+    - /usr/local/bin/etcd
+    - --name=recovery-etcd
+    - --listen-client-urls=http://0.0.0.0:32379
+    - --listen-peer-urls=http://0.0.0.0:32380
+    - --advertise-client-urls=http://localhost:32379
+    - --data-dir=/var/etcd/recovery
+    volumeMounts:
+      - mountPath: /var/etcd
+        name: etcd
+        readOnly: false
+  hostNetwork: true
+  restartPolicy: Never
+  volumes:
+    - name: etcd
+      emptyDir: {}
+    - name: etcdbackup
+      hostPath:
+        path: {{ .BackupDir }}
+`)
+
+var BootFromBackupEtcdTemplate = []byte(`Version: v1
+kind: Pod
+metadata:
+  name: bootstrap-etcd
+  namespace: kube-system
+  labels:
+    k8s-app: boot-etcd
+spec:
+  initContainers:
+  - name: recovery
+    image: {{ .Image }}
+    command:
+    - /bin/sh 
+    - -ec
+    - |
+      etcdctl snapshot restore \
+      /var/etcd-backupdir/{{ .BackupFile }} \
+      --data-dir=/var/etcd/data \
+      --name=boot-etcd \
+      --initial-cluster=boot-etcd=http://${{ .BootEtcdServiceIP }}:12380 \
+      --initial-cluster-token=bootkube \
+      --initial-advertise-peer-urls=http://${{ .BootEtcdServiceIP }}:12380 \
+      --skip-hash-check=true 
+    env:
+      - name: ETCDCTL_API
+        value: "3"
+    volumeMounts:
+        - mountPath: /var/etcd
+          name: etcd
+          readOnly: false
+        - mountPath: /var/etcd-backupdir
+          name: etcdbackup
+          readOnly: false
+  - name: cleanup
+    image: {{ .Image }}
+    command:
+    - /bin/sh 
+    - -ec
+    - |
+      /usr/local/bin/etcd \
+      --listen-client-urls=http://0.0.0.0:32379 \
+      --listen-peer-urls=http://0.0.0.0:32380 \
+      --advertise-client-urls=http://localhost:32379 \
+      --data-dir=/var/etcd/data && \
+      etcdctl \
+      --endpoints=http://localhost:32379 \
+      del /registry/ThirdPartyResourceData/etcd.coreos.com/clusters/kube-system/kube-etcd
```

Comment:
My original understanding was that you would:
(1) Start temporary etcd from backup, creating a datadir and extracting the manifests
(2) The bootstrap etcd would talk to the datadir you created in (1)

Thus, I thought you'd be able to delete the TPR from the running etcd in (1). However, it looks like (2) also starts up directly from the backup? In that case then I agree that there's no place in code to remove the TPR, so I suppose this should work. I'd say let's go ahead and try it.